ex2.py

```diff
@@ -0,0 +1,33 @@
+from functools import wraps
+
+
+def type_check(correct_type):
+    def type_check_decorator(func):
+        @wraps(func)
+        def wrapper(arg):
+            if not isinstance(arg, correct_type):
+                raise InvalidTypeError(f"Expected: {correct_type}.")
+            return func(arg)
+        return wrapper
+    return type_check_decorator
+
+
+class InvalidTypeError(Exception):

```

Comment:
It's bad practice to not implement a class.
Read a bit about how to implement a custom error class and do it.